lib/logstash/inputs/redis.rb

```diff
@@ -107,28 +107,26 @@ def is_list_type?
 
   # private
   def redis_params
+    params = {
+        :timeout => @timeout,
+        :db => @db,
+        :password => @password.nil? ? nil : @password.value,
+        :ssl => @ssl
+    }
+
     if @path.nil?
-      connectionParams = {
-        :host => @host,
-        :port => @port
-      }
+      params[:host] = @host
+      params[:port] = @port
     else
       @logger.warn("Parameter 'path' is set, ignoring parameters: 'host' and 'port'")
-      connectionParams = {
-        :path => @path
-      }
+      params[:path] = @path
     end
 
-    baseParams = {
-      :timeout => @timeout,
-      :db => @db,
-      :password => @password.nil? ? nil : @password.value,
-      :ssl => @ssl
-    }
-
-    return connectionParams.merge(baseParams)
+    params
   end
 
+  TIMEOUT = 5 # Redis only supports Integer values
```

Comment:
this timeout is internal for the redis (read/write) operation, we catch `TimeoutError` and retry in a loop,
provides a way for interrupting the input.

Reply:
Since we have multiple timeouts here in layers, it would be helpful to name this something distinct.

```suggestion
  BLOCKING_CALL_TIMEOUT = 5 # Redis only supports Integer values
```